File: src/lessons/welcome/array/IndexOfMaxValueEntity.java
package lessons.welcome.array;

import jlm.universe.array.ArrayEntity;
import jlm.universe.array.ArrayWorld;
import lessons.welcome.array.IndexOfMaxValueEntity;

public class IndexOfMaxValueEntity extends ArrayEntity {

	@Override
	public void run() {
		ArrayWorld w = (ArrayWorld) this.getWorld();
		this.result = this.indexOfMaximum(w.getValues());
	}

	/* BEGIN TEMPLATE */
// computes the index of the maximum of the values contained in tab variable
public int indexOfMaximum(int[] tab) {
	/* BEGIN SOLUTION */
	int max = Integer.MIN_VALUE;
	int index = 0;
	for (int i=0; i<tab.length; i++) {
		if (tab[i] >= max) {
			max = tab[i];
			index = i;
		}
	}
	return index;
	/* END SOLUTION */
}

	/* END TEMPLATE */

}

File: resources/exercises/array/search/IndexOfMaxValueEntity.py
# BEGIN TEMPLATE
def indexOfMaxValue(nums):
  # BEGIN SOLUTION
  max=nums[0]
  maxIdx = 0
  for i in range(len(nums)):
    if nums[i]>max:
      max = nums[i]
      maxIdx = i
  return maxIdx

# END SOLUTION
# END TEMPLATE
